apps/token-manager/app/src/hooks.js

```diff
@@ -0,0 +1,36 @@
+import { useRef, useEffect, useCallback } from 'react'
+
+export function useClickOutside(cb, existingRef) {
+  const ref = existingRef || useRef()
```

Comment:
When would one not pass an `existingRef`? 

I'm a bit confused about the utility of `ref` when not passing an `existingRef`

Reply:
`useClickOutside` can use an existing `ref` or return a new one:

```JavaScript
// creates new ref
const {ref}=useClickOutside(handleCb)
<input type="text" ref={ref} />

// existing
const {ref}=useClickOutside(handleCb, existingRef)
// ref and existingRef would be the same, but no need to consume ref anymore
useClickOutside(handleCb, existingRef)
```